MoltenVK/MoltenVK/GPUObjects/MVKDevice.mm

```diff
@@ -670,7 +679,12 @@
 	}
 
 	uint mtlFmtsCnt = sizeof(mtlFormats) / sizeof(MTLPixelFormat);
+#if MVK_MACOS
```

Comment:
I know this is not new in this update...but the way that the dynamic nature of the value of `mtlFmsCnt` is used to define indexing into `mtlFormats` depends on the last consecutive elements of `mtlFormats` being the only ones that are dynamically elided. 

This works for now, but feels fragile. For example...if a device in the future removed any except the last of the four `XR` formats...it would trigger undefined behaviour.

Reply:
Fixed in #943.